secv_guis/bimask_app/main_window.py

```diff
@@ -287,23 +207,31 @@ def new_image(self, img_path, initial_mask_color=(219, 54, 148, 150),
         self.saved_state_tracker = SavedStateTracker()
         return True
 
-    def preannot_from_path(self, preannot_path, rgba, keep_p_value=0.05,
-                           discard_p_value=0.5, normalize=False):
+    def preannot_from_path(self, preannot_path, rgba, upper_thresh=100,
+                           lower_thresh=90, normalize=False):
         """
         This method is prototype-ish: It loads an ``.npz`` file with and
         'entropy' field, expected to have a numpy float matrix with same
-        shape as the image.
+        shape as the image. Alternatively it takes a greyscale image file 
+        suppoted by PIL.
         """
         assert self.scene().img_pmi is not None, \
             "You need to load an image first!"
-        im = Image.open(preannot_path)
-        self._preannot_pmap = np.asarray(im)
+        if preannot_path.endswith(".npz") or preannot_path.endswith(".npy"):
+            self._preannot_pmap = np.load(preannot_path)["entropy"]
```

Comment:
Eventually we want to get rid of this and do things your way. The "entropy" bit is just completely ad-hoc and useless at this point

Reply:
I could remove it, but I thought you may need it for legacy reasons